src/common.cpp

```diff
@@ -240,20 +242,23 @@ namespace irods
         using irods::experimental::filesystem::perms;
 
         switch (_p) {
+            // clang-format off
             case perms::null:            return "null";
             case perms::read_metadata:   return "read_metadata";
-            case perms::read_object:     return "read_object";
-            case perms::read:            return "read_object";
+            case perms::read_object:
+            case perms::read:            return "read";
```

Comment:
still feeling like we can wrap these two (and the two for `write`) in a `enable_4_2_compatibility` conditional...

```
if enable_4_2_compatibility
    read
else
    read_object
```

Reply:
Agreed on the apps *using* the HTTP API... but I think we optimize for 4.3+.   This 4.2 layer is a workaround/compromise for a couple years until everyone is on 4.3+.

I'd rather have a box around the 'old' names and use the new names wherever possible.   It's only not possible here everywhere because 4.2 doesn't understand the new ones, so we make a workaround until they upgrade.  We can write that somewhere in the README for this compatibility layer so a developer/admin can make an appropriate assessment about where they want to spend their effort/time.

Reply:
With a little bit of work, we can make the HTTP API accept both forms of input and present only the 4.3 names when sending a response.

Reply:
that could work.   any app that uses this will, by definition, be new... so, that seems okay to always send the future-proof strings back.

Reply:
The application now accepts 4.3 and 4.2 names for permissions, but only returns the 4.3 names.